File: Roll a Ball/Assets/Scripts/PlayerController.js
#pragma strict

import UnityEngine.UI;


private var rb : Rigidbody;
private var count : int;

var speed : float;
var countText : Text;
var winText : Text;

function Start () {
	rb = GetComponent.<Rigidbody>();
	count = 0;
	SetCountText();
	winText.text = '';
}

function FixedUpdate () {
	var moveHorizontal : float = Input.GetAxis("Horizontal");
	var moveVertical : float = Input.GetAxis("Vertical");

	var movement : Vector3 = new Vector3 (moveHorizontal, 0, moveVertical);

	rb.AddForce(movement * speed);
}

function OnTriggerEnter (other : Collider) {
	if (other.gameObject.CompareTag("Pickup")) {
		other.gameObject.SetActive(false);
		count += 1;
		SetCountText();
	}
}

function SetCountText() {
	countText.text = "Count: " + count;
	if (count >= 8) {
		winText.text = 'You win!';
	}
}

File: 2D_UFO/Assets/Scripts/PlayerController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class PlayerController : MonoBehaviour {

    public float speed;
    public Text countText;
    public Text winText;
    public int winCount;

    private Rigidbody2D rb2d;
    private int count;

    private void Start()
    {
        rb2d = GetComponent<Rigidbody2D>();
        count = 0;
        winText.text = "";
        SetCountText();
    }

    private void FixedUpdate()
    {
        float moveHorizontal = Input.GetAxis("Horizontal");
        float moveVertical = Input.GetAxis("Vertical");
        Vector2 movement = new Vector2(moveHorizontal, moveVertical);
        rb2d.AddForce(movement*speed);
    }

    private void OnTriggerEnter2D(Collider2D otherCollider)
    {
        if (otherCollider.gameObject.CompareTag("PickUp")) {
            otherCollider.gameObject.SetActive(false);
            count = count + 1;
            SetCountText();
        }
    }

    void SetCountText()
    {
        countText.text = "Count: " + count.ToString();
        if (count >= winCount)
        {
            winText.text = "You Win!";
        }
    }
}
